carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/DriverHelper.java

```diff
@@ -1120,8 +1120,12 @@ public boolean waitUntil(ExpectedCondition<?> condition, long timeout) {
 		        .ignoring(WebDriverException.class)
 				.ignoring(NoSuchSessionException.class);
 		try {
+		    long startMillis = System.currentTimeMillis();
 			wait.until(condition);
 			result = true;
+            if ((System.currentTimeMillis() - startMillis) > 2 * RETRY_TIME) {
```

Comment:
in 2 times bigger then timeout!
"> 2 * timeout"

Reply:
Also, this piece of code should be executed in finally block to capture positive and negative cases